### ISSUE TITLE
Resume on App Startup 

### ISSUE THREAD

**Issue by HankG** [supabase-community/supabase-flutter#174]
Under the pre-1.0 API resume on app launch happened by calling into the SupabaseAuthState.recoverSupabaseSession(). A version of that method still exists on SupabaseAuth but it is now private and only accessed through SupabaseAuth.didChangeAppLifecycleState with a "resumed" AppLifecycleState value. I can't see where that is being called. In the Supabase Auth UI project I created a method which calls `SupabaseAuth.instance.didChangeAppLifecycleState(AppLifecycleState.resumed)`. I can get it to resume the session but there are a couple of problems. First, the _recoverSupabaseSession() is async but didChangeAppLifecycleState is not so there is no way to await for the return or to know what the return value was. Second, it feels like this is supposed to be initiated somewhere else but I can't determine where. My workaround right now is to call this right after init, add a button to a splash screen to go into the real part of the app, and then only click that button once I see that the session has been restored in the output. That is all obviously not ideal.

Is there a plan to have this reload exposed elsewhere in a way we can wait for it at our app level or expose the recover method itself for direct initiation?

**Comment by bdlukaa** [comment]
The session is automatically restored when calling `Supabase.initialize`.

For which use case would you want to restore it manually?

**Comment by HankG** [comment]
I don't want to restore it manually. It's not being restored automatically right now in the 1.x version of the library.

**Comment by bdlukaa** [comment]
What's the output when you print `await SupabaseAuth.initialSession`?

**Comment by HankG** [comment]
I'm using `SuperbaseAuthUi().initSupabase()` which is returning null. I'm not sure what "await" is doing there in their example code since `initSupabase` isn't an async function and it doesn't return the result of `Supabase.initialize` which is a Future<Supabase>...hmm...

**Comment by HankG** [comment]
Yes on that aspect unless we want the main method locking on trying to load from the local store. Secondarily I can't figure out the best idiomatic place to put a `Supabase.instance.client.auth.onAuthStateChange` to track the state changes that used to get picked up by the SupabaseAuthState class. Handling it in the context of the MyApp causes an error `Unhandled Exception: Navigator operation requested with a context that does not include a Navigator.` since of course the MaterialApp is what has the navigation. What's the best place for it?

**Comment by bdlukaa** [comment]
@HankG SupabaseAuthState has been removed. https://github.com/supabase-community/supabase-flutter#authentication
